File: ecs-java-s3-workshop/src/main/java/com/emc/ecs/s3/sample/_07_PresignedURL.java
/*
 * Copyright 2013 EMC Corporation. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.txt
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.emc.ecs.s3.sample;

import com.emc.object.s3.S3Client;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.URL;
import java.util.Date;

public class _07_PresignedURL {

	public static void main(String[] args) throws Exception {
    	// create the ECS S3 Client
    	S3Client s3 = ECSS3Factory.getS3Client();

    	// retrieve the key value from user
        System.out.println( "Enter the object key:" );
        String key = new BufferedReader( new InputStreamReader( System.in ) ).readLine();
        
        // retrieve the expiration time for the object from user
        System.out.print( "How many hours should this tag be valid? " );
        String hours = new BufferedReader( new InputStreamReader( System.in ) ).readLine();

        // convert hours to a date
        Date expiration = new Date();
        long curTime_msec = expiration.getTime();
        long nHours = Long.valueOf(hours);
        curTime_msec += 60 * 60 * 1000 * nHours;
        expiration.setTime(curTime_msec); 
        
        // generate the object's pre-signed URL
        URL url = s3.getPresignedUrl(ECSS3Factory.S3_BUCKET, key, expiration);

        
        // print object's pre-signed URL
    	System.out.println( String.format("object [%s/%s] pre-signed URL: [%s]",
    			ECSS3Factory.S3_BUCKET, key, url.toString()));
    }
}

File: ecs-net-workshop/aws-net-workshop/examples/_07_PresignedURL.cs
using System;
using ECSSDK.S3;
using Amazon.S3.Model;
using Amazon.S3;

namespace ecs_net_workshop.examples
{
    class _07_PresignedURL
    {
        public static void Main(string[] args)
        {
            // create the ECS S3 client
            ECSS3Client s3 = ECSS3Factory.getS3Client();

            // retrieve the object key and hours before expiration
            Console.Write("Enter the object key: ");
            string key = Console.ReadLine();
            Console.Write("How long should this tag be valid?: ");
            string hours = Console.ReadLine();

            // create expiration based on input
            DateTime expiration = DateTime.Now.AddHours(Convert.ToDouble(hours));

            // create the request object to generate pre-signed url
            GetPreSignedUrlRequest request = new GetPreSignedUrlRequest()
            {
                BucketName = ECSS3Factory.S3_BUCKET,
                Key = key,
                Expires = expiration,
                Verb = HttpVerb.GET,
            };

            // get pre-signed url
            string url = s3.GetPreSignedURL(request);

            // print objects pre-signed url
            Console.WriteLine(string.Format("Object {0}/{1} pre-signed url: {2}", ECSS3Factory.S3_BUCKET, key, url));
            Console.ReadLine();
        }
    }
}
